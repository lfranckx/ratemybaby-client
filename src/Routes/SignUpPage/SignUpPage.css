#register-section {
    display: inline-block;
    background-color: white;
    border-radius: 10px;
    padding: 40px;
    box-shadow: 1.5px 3px 5px grey;
    color: rgb(41, 39, 39);
    width: 400px;
}

#register-header {
    font-size: 2.6rem;
    opacity: 0.9;    
    margin-bottom: 40px
}

#register-section label {
    color: rgb(41, 39, 39);
}

#sign-up-button {
    margin-top: 30px;
    background-color: rgb(249, 103, 103);
    background-image: linear-gradient(to right, rgb(247, 87, 87) , rgb(249, 147, 103));    
    display: inline-block;
    margin-top: 10px;
    padding: 10px 30px;
    border-radius: 50px;
    border: none;
    color: white;
    cursor: pointer;
}