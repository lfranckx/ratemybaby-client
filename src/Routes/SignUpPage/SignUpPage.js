import React, { Component } from 'react'
import SignUpForm from '../../Components/SignUpForm/SignUpForm'
import './SignUpPage.css'

export default class SignUpPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  }

  handleSignUpSuccess = () => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/createprofile'
    history.push(destination)  }

  render() {
    return (
      <section id="register-section">
        <h3 id="register-header">Register</h3>
        <SignUpForm
          onSignUpSuccess={this.handleSignUpSuccess}
        />
      </section>
    )
  }
}
