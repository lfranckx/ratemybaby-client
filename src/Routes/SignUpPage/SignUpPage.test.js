import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SignUpPage from './SignUpPage';

describe(`SignUpPage component`, () => {

    it(`renders sign up page by default`, () => {
        const wrapper = shallow(<SignUpPage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
})