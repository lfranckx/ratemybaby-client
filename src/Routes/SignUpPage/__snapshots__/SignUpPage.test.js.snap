// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpPage component renders sign up page without crashing 1`] = `
<section
  id="register-section"
>
  <h2
    id="register-header"
  >
    Register
  </h2>
  <SignUp
    onRegistrationSuccess={[Function]}
    onSignUpSuccess={[Function]}
  />
</section>
`;
